### PR TITLE
Fix and modernize our github templates

### PR DESCRIPTION
**Pull request by Generalcamo**
NUFC

Fixes the template committed in #16170. Also adds an issue config YML file, to direct suggestions and non-code issues to their relevant areas.

### CODE REVIEW

**Comment by Generalcamo** [comment]
!review

**Comment by Generalcamo** [comment]
!bugfix

**Review by Arrow768 — APPROVED** [review]

**Review by alsoandanswer — APPROVED** [review]

**Review by JohnWildkins — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/BUG-REPORT.yaml [hunk 1]
```diff
@@ -1,52 +1,77 @@
...
+  attributes:
+    label: How to reproduce
+    description: Describe how to reproduce the issue or, if unable to reproduce it, what steps you did that you believe might have caused the bug.
+    placeholder: |
+      Tell us the steps required to trigger your bug. Example:
+      1. Open the PDA
+      2. Click the "Crew Manifest" program
+      3. Byond crashes
+  validations:
+    required: false
```
Comment (FluffyGhoster) [inline .github/ISSUE_TEMPLATE/BUG-REPORT.yaml:59]:
```suggestion
    required: true
```
A bug report that has no reproduction and no idea of even what action might have caused it is essentially not reproducible and thus unfixable, which is why it was set as required.

File: .github/ISSUE_TEMPLATE/BUG-REPORT.yaml [hunk 2]
```diff
@@ -1,52 +1,77 @@
...
+    options:
+      - Horizon
+      - Intrepid
+      - Spark
+      - Away Site
+      - Third party ship
+      - Overmap
+      - Event / Other
+  validations:
+    required: false
```
Comment (FluffyGhoster) [inline .github/ISSUE_TEMPLATE/BUG-REPORT.yaml:37]:
```suggestion
    required: true
```

**Review by FluffyGhoster — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/BUG-REPORT.yaml (hunk 1 above)
Comment (Generalcamo) [reply .github/ISSUE_TEMPLATE/BUG-REPORT.yaml:59]: This is not always known. Forcing people to write this information is not productive, especially if they just put in "I dunno" or something.

File: .github/ISSUE_TEMPLATE/BUG-REPORT.yaml (hunk 2 above)
Comment (Generalcamo) [reply .github/ISSUE_TEMPLATE/BUG-REPORT.yaml:37]: This one is fair, I will change this.

File: .github/ISSUE_TEMPLATE/BUG-REPORT.yaml (hunk 1 above)
Comment (FluffyGhoster) [reply .github/ISSUE_TEMPLATE/BUG-REPORT.yaml:59]: I disagree, putting "I don't know" is a willing action after the person thinks and writes it, far more thoughtful that ignoring the question or just go "whatever, I don't want to think about it so I leave it blank, next".

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**